.github/workflows/tests.yaml

```
@@ -174,23 +175,57 @@ jobs:
         if: steps.filter.outputs.ignore == 'false'
         run: df -h
 
-      # Build all targets first to isolate build failures.
-      - name: Build (${{ matrix.build_mode }})
-        if: steps.filter.outputs.ignore == 'false'
+      # Build and run all targets on branch pushes to ensure we always have a
+      # clean tree. We don't expect this to be an interactive path and so don't
+      # optimize the latency of this step.
+      - name: Test (${{ matrix.build_mode }})
+        if:
+          steps.filter.outputs.ignore == 'false' && github.event_name == 'push'
         env:
           # 'libtool_check_unique failed to generate' workaround.
           # https://github.com/bazelbuild/bazel/issues/14113#issuecomment-999794586
           BAZEL_USE_CPP_ONLY_TOOLCHAIN: 1
-        run: bazelisk build -c ${{ matrix.build_mode }} //...:all
+        run: |
+          bazelisk test -c ${{ matrix.build_mode }} //...
 
-      # Run all test targets.
+      # Compute the set of possible rules impacted by this change using
+      # Bazel-based diffing. This lets PRs and the merge queue have a much more
+      # efficient test CI action by avoiding even enumerating (and downloading)
+      # all of the unaffected Bazel targets
```

```suggestion
      # all of the unaffected Bazel targets.
```

Sorry, punctuation style nit.